lib/ex_doc/cli.ex

```diff
@@ -35,6 +36,11 @@ defmodule ExDoc.CLI do
     else
       generate(args, opts, generator)
     end
+
+    if List.keymember?(opts, :warnings_as_errors, 0) and ExDoc.WarningCounter.count() > 0 do
+      IO.warn("Doc generation failed due to warnings while using the --warnings-as-errors option")
```

Comment:
I don't think it should be printed as a warning (`warning: `) so I'd consider just a simple IO.puts.

```suggestion
      IO.puts("Doc generation failed due to warnings while using the --warnings-as-errors option")
```